### PR TITLE
Fix PackerManager benchmark and optimize hashing.Writer

### PR DESCRIPTION
**Pull request by greatroar**
What is the purpose of this change? What does it change?
--------------------------------------------------------

I was trying to see whether a few potential optimizations to hashing.Writer would make it faster. Before I could do that, I had to fix the PackagerManager benchmark to be more realistic.

The only actual optimization that had a (very minor) effect and didn't make the code unreadable is included as the third commit.

Was the change discussed in an issue or in the forum before?
------------------------------------------------------------

No.

Checklist
---------

- [x] I have read the [Contribution Guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches)
- [x] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
- [x] I have added tests for all changes in this PR
- [ ] I have added documentation for the changes (in the manual)
- [ ] There's a new file in `changelog/unreleased/` that describes the changes for our users (template [here](https://github.com/restic/restic/blob/master/changelog/TEMPLATE))
- [x] I have run `gofmt` on the code in all commits
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits)
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

File: internal/hashing/writer.go [hunk 1]
```diff
@@ -15,13 +15,14 @@ type Writer struct {
...
 		h: h,
-		w: io.MultiWriter(w, h),
+		w: w,
 	}
 }
 
 // Write wraps the write method of the underlying writer and also hashes all data.
 func (h *Writer) Write(p []byte) (int, error) {
 	n, err := h.w.Write(p)
+	h.h.Write(p[:n])
```
Comment (MichaelEischer) [inline internal/hashing/writer.go]: The Hash interface states that a call to `Write()` never returns an error. Does this also apply to the number of written bytes?

Comment (greatroar) [reply internal/hashing/writer.go]: I'm not exactly sure if this is what you mean, but io.Writer "must return a non-nil error if it returns n < len(p)". So if a hash.Writer never returns an error, it must always return the length of its argument.

Comment (MichaelEischer) [reply internal/hashing/writer.go]: That's exactly the bit of information I've missed. Together with "hash.Write" never returns an error this implies that all n bytes are always written.

**Review by MichaelEischer — APPROVED** [review]
LGTM

**Pull request merged**